prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java

```diff
@@ -94,14 +94,12 @@ public PhysicalPlan visitIndexAggregation(PrometheusLogicalMetricAgg node,
   public PhysicalPlan visitRelation(LogicalRelation node,
                                     PrometheusMetricScan context) {
     PrometheusMetricTable prometheusMetricTable = (PrometheusMetricTable) node.getTable();
-    if (prometheusMetricTable.getMetricName() != null) {
```

Comment:
Just my feel that it's a little hard to follow why several `if` check removed in this PR. it's fine if it's covered in UT.

Reply:
In earlier implementation, query_range function implementation got coupled with normal PPL queries source = my_prometheus.proemtheus_http_requests_total implementation. This PR at least decouples few of the things.

Need to refactor the existing implementation. I tried to refactor in this PR, the changes are becoming huge. Will cover them in another PR.

Reply:
With the new implementation. Visit Relation will always have metricName.

Reply:
I see. Please add TODO comment or put all refactor items in Github issue. Thanks!

Reply:
https://github.com/opensearch-project/sql/issues/1884